datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1601,12 +1647,17 @@ fn from_substrait_named_struct(
     );
     if name_idx != base_schema.names.len() {
         return substrait_err!(
-                                "Names list must match exactly to nested schema, but found {} uses for {} names",
-                                name_idx,
-                                base_schema.names.len()
-                            );
+            "Names list must match exactly to nested schema, but found {} uses for {} names",
+            name_idx,
+            base_schema.names.len()
+        );
+    }
+    let mut df_schema = DFSchema::try_from(Schema::new(fields?))?;
+    match field_qualifier {
+        None => (),
+        Some(fq) => df_schema = df_schema.replace_qualifier(fq),
     }
-    Ok(DFSchemaRef::new(DFSchema::try_from(Schema::new(fields?))?))
+    Ok(DFSchemaRef::new(df_schema))
```

Comment:
If this is made public, I think it might make sense to return a `DFSchema` instead of a `DFSchemaRef` to make it more flexibly for users, whom can choose to wrap the DFSchema if they need it.